group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/AssignmentTopicMetadata.java

```diff
@@ -16,44 +16,83 @@
  */
 package org.apache.kafka.coordinator.group.assignor;
 
-/**
- * Metadata of a topic.
- */
-public class AssignmentTopicMetadata {
+import org.apache.kafka.common.Uuid;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class AssignmentTopicMetadata implements AssignmentTopicDescriber {
```

Comment:
TargetAssignmentBuilder is part of the consumer package though and I don't think this has anything to do with the consumer except for the fact that its the topics that the consumer is subscribed to. Is this cause we're gonna make the assignor package public and this class isnt?

Reply:
I see it in the consumer package because this class is about how the TargetAssignmentBuilder uses the PartitionAssignor interface and not really about the PartitionInterface itself. That's right, that assignor package will eventually become public.